frontend/src/utils/budgetUtil.ts

```diff
@@ -0,0 +1,158 @@
+import { Category, Transaction } from 'src/types/budget';
+
+export interface MonthlyDepositWithdraw {
+  [month: string]: DepositWithdraw;
+}
+
+export interface DepositWithdraw {
+  deposit: number;
+  withdraw: number;
+}
+
+export interface CategoryMap {
```

Comment:
It makes me think this type is a map with key as categoryId and category as value. Let's avoid this type of name unless it's really a map of category.

Reply:
Changed.